syntax = "proto3";

package ingestion;

option go_package = ".;proto_gen";



// -----------------------------------------------------------------------------
// Internal Services
// -----------------------------------------------------------------------------

// InternalAuthService provides a private API for the ingestion-service to
// validate API keys.
service InternalAuthService {
  // ValidateApiKey checks if an API key is valid.
  rpc ValidateApiKey(ValidateApiKeyRequest) returns (ValidateApiKeyResponse) {}
}

message ValidateApiKeyRequest {
  // The API key to be validated.
  string api_key = 1;
}

message ValidateApiKeyResponse {
  // Whether the API key is valid.
  bool is_valid = 1;
}
